jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorMergerTest.java

```diff
@@ -494,13 +516,35 @@ public void testDeploymentDesciptorMergeMergeCollectionsAvoidDuplicatesNamedObje
 		assertEquals(0, outcome.getConfiguration().size());
 		assertEquals(0, outcome.getEnvironmentEntries().size());
 		assertEquals(0, outcome.getEventListeners().size());
-		assertEquals(0, outcome.getGlobals().size());		
+		assertEquals(0, outcome.getGlobals().size());
 		assertEquals(0, outcome.getTaskEventListeners().size());
 		assertEquals(1, outcome.getWorkItemHandlers().size());
-		
+
 		// let's check if the slave version is preserved
 		NamedObjectModel model = outcome.getWorkItemHandlers().get(0);
 		assertEquals("Log", model.getName());
 		assertEquals("new org.jbpm.process.instance.impl.demo.CustomSystemOutWorkItemHandler()", model.getIdentifier());
 	}
+
+	private static final String jarLocRegexStr = "file:/.*/([\\d\\.]{3})\\.\\d(-SNAPSHOT)?/.*";
+	private static final Pattern jarLocRegex = Pattern.compile(jarLocRegexStr);
+
+	/**
+	 * This test will fail in the IDE because of the IDE will mess with the classpath.
+	 */
+	@Test
+	public void changeDefaultLimitSerializationClassesValueToTrueIn7x() {
+	    String codeLocStr = DeploymentDescriptor.class.getProtectionDomain().getCodeSource().getLocation().toString();
+	    Assume.assumeTrue("Test will not succeed in an IDE", codeLocStr.endsWith(".jar"));
+
+	    Matcher matcher = jarLocRegex.matcher(codeLocStr);
+	    assertTrue( "Fix regular expression: " + jarLocRegexStr, matcher.matches() );
+	    double jarVersion = Double.parseDouble(matcher.group(1));
+
+	    DeploymentDescriptorImpl depDesc = new DeploymentDescriptorImpl();
+
+	    assertTrue( "The default value of 'limitSerializationClasses is FALSE in 6.x and TRUE in 7.x",
+	            jarVersion < 7.0d && ! depDesc.getLimitSerializationClasses()
+	            || jarVersion >= 7.0d && depDesc.getLimitSerializationClasses() );
```

Comment:
Github is having problems displaying the actual code in the PR: see https://github.com/droolsjbpm/jbpm/pull/356/files#diff-d4f4a2a33edb3e2f281c94f1dfa8d522R539